packages/ckeditor5-list/tests/liststyleediting.js

```diff
@@ -923,6 +973,97 @@ describe( 'ListStyleEditing', () => {
 			} );
 		} );
 
+		describe( 'delete + undo', () => {
+			let editor, model, view;
+
+			beforeEach( () => {
+				return VirtualTestEditor
+					.create( {
+						plugins: [ Paragraph, ListStyleEditing, Typing, UndoEditing ]
+					} )
+					.then( newEditor => {
+						editor = newEditor;
+						model = editor.model;
+						view = editor.editing.view;
+					} );
+			} );
+
+			afterEach( () => {
+				return editor.destroy();
+			} );
+
+			// See: #7930.
+			it( 'should restore proper list style attribute after undo merging lists', () => {
```

Comment:
This test doesn't fail for implementation that is on `master` so it doesn't protect us from a regression.

Reply:
That's weird. I figured out that the bug (#7930) was fixed (accidentally) while I was working on the merging lists ticket (#7879).

Bisect showed me this commit – https://github.com/ckeditor/ckeditor5/commit/7aa952823a8b182dc41075fa8cf4cc3a452eb78b – as a point where the bug is no longer reproducible.

But on the parent commit – https://github.com/ckeditor/ckeditor5/commit/7aa952823a8b182dc41075fa8cf4cc3a452eb78b – I was able to reproduce the error (#7930).

I also checked the unit test – it behaves as expected. Passes for the first commit and fails for the second one.

Reply:
In case it was fixed in another PR it make sense that it passes now. I have verified that it indeed passes since 7aa9528.